modin/experimental/core/execution/native/implementations/hdk_on_native/partitioning/partition_manager.py

```diff
@@ -178,7 +178,13 @@ def _get_unsupported_cols(cls, obj):
 
             try:
                 at = pyarrow.Table.from_pandas(obj, preserve_index=False)
-            except (pyarrow.lib.ArrowTypeError, pyarrow.lib.ArrowInvalid) as err:
+            except (
+                pyarrow.lib.ArrowTypeError,
+                pyarrow.lib.ArrowInvalid,
+                ValueError,
+            ) as err:
+                if (type(obj) == ValueError) and obj.columns.is_unique:
+                    raise err
                 regex = r"Conversion failed for column ([^\W]*)"
                 unsupported_cols = []
                 for msg in err.args:
```

Comment:
would that `for` loop still work correctly for `ValueError`?

Reply:
It works, but the regex does not match anything, because all columns are mangled with F_. In this case, all columns are returned as unsupported.